packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/ScanElementsCollector.java

```diff
@@ -40,8 +42,9 @@ public ScanElementsCollector collect() {
                 .collect(Collectors.toList());
 
         entities = endpoints.stream()
-                .flatMap(cls -> cls.getInheritanceChainStream()
-                        .flatMap(ClassInfoModel::getDependenciesStream))
+                .flatMap(cls -> Streams.combine(cls.getInheritanceChainStream(),
+                        cls.getInterfacesStream()))
```

Comment:
Looks like this is not covered by tests. It is somewhat related to the upcoming test for `@EndpointExposed`: https://github.com/vaadin/hilla/pull/520/commits/995d788445b43ce844deebacae207c7c68165d74

Reply:
True, but let us address it in https://github.com/vaadin/hilla/pull/520 maybe?